"Failed to solve: process ... exit code 2" when building demo from source with Docker

# Bug Report

OpenTelemetry Demo 1.4

## Symptom

Since yesterday evening, 2023-05-09, I'm getting the following error message when building the demo from source with `docker compose build`:

`
failed to solve: process "/bin/sh -c git clone https://github.com/open-telemetry/opentelemetry-cpp \t&& cd opentelemetry-cpp/     \t&& git checkout tags/v${OPENTELEMETRY_CPP_VERSION} -b v${OPENTELEMETRY_CPP_VERSION} \t&& mkdir build \t&& cd build \t&& cmake .. -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DWITH_EXAMPLES=OFF -DWITH_OTLP=ON -DWITH_OTLP_GRPC=ON \t&& make -j$(nproc || sysctl -n hw.ncpu || echo 1) install && cd ../.. && rm -rf opentelemetry-cpp" did not complete successfully: exit code: 2
`

## Additional Context

I tried it on Windows and macOS.

From error
```
#0 6.814                  from /opentelemetry-cpp/build/generated/third_party/opentelemetry-proto/opentelemetry/proto/collector/logs/v1/logs_service.grpc.pb.cc:6:
#0 6.814 /usr/include/absl/base/policy_checks.h:79:2: error: #error "C++ versions less than C++14 are not supported."
#0 6.814    79 | #error "C++ versions less than C++14 are not supported."
```
It seems Alpine 3.18 brings latest version of abseil/gRPC packages which needs C++14 compiler to build. Can you try build by adding `-DCMAKE_CXX_STANDARD=14` as CMake argument here -

https://github.com/open-telemetry/opentelemetry-demo/blob/02a91d4abeaad0aed2e9957ffe82da64b600159d/src/currencyservice/Dockerfile#L29

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,8 @@ release.
   ([#878](https://github.com/open-telemetry/opentelemetry-demo/pull/878))
 * [kafka] remove KRaft mode support workarounds
   ([#880](https://github.com/open-telemetry/opentelemetry-demo/pull/880))
+* [currencyservice] Fix OTel C++ build and update OTel version to 1.9.0
+  ([#886](https://github.com/open-telemetry/opentelemetry-demo/pull/886))
 
 ## 1.4.0
 

--- a/src/currencyservice/Dockerfile
+++ b/src/currencyservice/Dockerfile
@@ -19,14 +19,16 @@ FROM alpine as builder
 RUN apk update
 RUN apk add git cmake make g++ grpc-dev re2-dev protobuf-dev c-ares-dev
 
-ARG OPENTELEMETRY_CPP_VERSION=1.8.2
+ARG OPENTELEMETRY_CPP_VERSION=1.9.0
 
 RUN git clone https://github.com/open-telemetry/opentelemetry-cpp \
 	&& cd opentelemetry-cpp/ \
     	&& git checkout tags/v${OPENTELEMETRY_CPP_VERSION} -b v${OPENTELEMETRY_CPP_VERSION} \
 	&& mkdir build \
 	&& cd build \
-	&& cmake .. -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DWITH_EXAMPLES=OFF -DWITH_OTLP=ON -DWITH_OTLP_GRPC=ON \
+	&& cmake .. -DCMAKE_CXX_STANDARD=17 -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+				-DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF \
+				-DWITH_EXAMPLES=OFF -DWITH_OTLP=ON -DWITH_OTLP_GRPC=ON \
 	&& make -j$(nproc || sysctl -n hw.ncpu || echo 1) install && cd ../.. && rm -rf opentelemetry-cpp
 
 COPY . /currencyservice

